engine/tests/test_newnode_neuro.py

```diff
@@ -0,0 +1,162 @@
+from ..submitter import Submitter
+
+#dj niworkflows vs ...??
+from nipype.interfaces.utility import Rename
+import nipype.interfaces.freesurfer as fs
+
+from fmriprep.interfaces.freesurfer import PatchedConcatenateLTA as ConcatenateLTA
+
+@pytest.fixture()
+def change_dir(request):
+    orig_dir = os.getcwd()
+    test_dir = os.path.join(orig_dir, "/nipype/nipype/pipeline/engine/test_neuro")
+    os.makedirs(test_dir, exist_ok=True)
+    os.chdir(test_dir)
+
+    def move2orig():
+        os.chdir(orig_dir)
+
+    request.addfinalizer(move2orig)
+
+import pdb
+
+Name = "example"
+DEFAULT_MEMORY_MIN_GB = None
+# TODO, adding fields to Inputs (subject_id)
+Inputs = {"subject_id": "sub-01",
+          "output_spaces": ["fsaverage", "fsaverage5"],
+          "source_file": "/fmriprep_test/workdir1/fmriprep_wf/single_subject_01_wf/func_preproc_ses_test_task_fingerfootlips_wf/bold_t1_trans_wf/merge/vol0000_xform-00000_merged.nii",
+          "t1_preproc": "/fmriprep_test/output1/fmriprep/sub-01/anat/sub-01_T1w_preproc.nii.gz",
+          "t1_2_fsnative_forward_transform": "/fmriprep_test/workdir1/fmriprep_wf/single_subject_01_wf/anat_preproc_wf/surface_recon_wf/t1_2_fsnative_xfm/out.lta",
+          "subjects_dir": "/fmriprep_test/output1/freesurfer/"
+}
+
+Plugins = ["serial", "mp", "cf", "dask"]
+
+def select_target(subject_id, space):
+    """ Given a source subject ID and a target space, get the target subject ID """
+    return subject_id if space == 'fsnative' else space
+
+@pytest.mark.parametrize("plugin", Plugins)
+def test_neuro(change_dir, plugin):
+
+    # wf = Workflow(name, mem_gb_node=DEFAULT_MEMORY_MIN_GB,
+    #               inputs=['source_file', 't1_preproc', 'subject_id',
+    #                       'subjects_dir', 't1_2_fsnative_forward_transform',
+    #                       'mem_gb', 'output_spaces', 'medial_surface_nan'],
+    #               outputs='surfaces')
+    #
+    #dj: why do I need outputs?
+
+
+    wf = NewWorkflow(name=Name, inputs=Inputs, workingdir="test_neuro_{}".format(plugin), print_val=False,
+                     wf_output_names=[("sampler", "out_file", "sampler_out"), ("targets", "out", "target_out")])
+
+    # @interface
+    # def select_target(subject_id, space):
+    #     """ Given a source subject ID and a target space, get the target subject ID """
+    #     return subject_id if space == 'fsnative' else space
+
+
+
+    # wf.add('targets', select_target(subject_id=wf.inputs.subject_id))
+    #   .map('space', space=[space for space in wf.inputs.output_spaces
+    #                        if space.startswith('fs')])
+
+    #dj: don't have option in map to connect with wf input
+
+    wf.add(runnable=select_target, name="targets", subject_id="subject_id", output_names=["out"],
+           out_read=True, print_val=False)\
+        .map_node(mapper="space", inputs={"space": [space for space in Inputs["output_spaces"]
+                                               if space.startswith("fs")]})
```

Comment:
@effigies - in the original function is there a reason why this is inside the workflow, as opposed to sending the workflow the appropriate list of spaces?

Reply:
To summarize IRL conversation: No reason that it couldn't be handled from outside this scope. But it is an opportunity to ensure that we do handle such workflow-writer decisions correctly. (Or ensure that we make it fail, if we want to disallow such.)